wallet/wallet.go

```diff
@@ -3716,6 +3717,276 @@ func (w *Wallet) ListUnspent(ctx context.Context, minconf, maxconf int32, addres
+
+			if output.Amount < minAmount {
+				log.Infof("Skipping utxo %s, amount: %s, min: %s", output.Hash.String(), output.Amount, minAmount)
+				continue
+			}
+
+			if inputSelectionMethod == types.OneUTXOInputSelection {
+				// We're selecting only one utxo so this loop will run until we find
+				// a single utxo that can pay the target amount.
+
+				if targetAmount > output.Amount {
+					// continue if this utxo cannot pay the require amount
+					continue
+				}
+			} else if inputSelectionMethod == types.UniqueTxInputSelection {
+				// skip duplicate tx id
+				_, seenTxID := seenTxIDs[output.Hash.String()]
+				if seenTxID {
+					log.Infof("Skipping duplicate txid: %s:%d", output.Hash.String(), output.Index)
+					continue
+				}
+			}
+
+			details, err := w.txStore.TxDetails(txmgrNs, &output.Hash)
+			if err != nil {
+				return err
+			}
+
+			// Outputs with fewer confirmations than the minimum are excluded.
+			confs := confirms(output.Height, tipHeight)
+			if confs < minconf {
+				continue
+			}
+
+			// Only mature coinbase outputs are included.
+			if output.FromCoinBase {
+				if !coinbaseMatured(w.chainParams, output.Height, tipHeight) {
+					continue
+				}
+			}
+
+			switch details.TxRecord.TxType {
+			case stake.TxTypeSStx:
+				// Ticket commitment, only spendable after ticket maturity.
+				if output.Index == 0 {
+					if !ticketMatured(w.chainParams, details.Height(), tipHeight) {
+						continue
+					}
+				}
+				// Change outputs.
+				if (output.Index > 0) && (output.Index%2 == 0) {
+					if !ticketChangeMatured(w.chainParams, details.Height(), tipHeight) {
+						continue
+					}
+				}
+			case stake.TxTypeSSGen:
+				// All non-OP_RETURN outputs for SSGen tx are only spendable
+				// after coinbase maturity many blocks.
+				if !coinbaseMatured(w.chainParams, details.Height(), tipHeight) {
+					continue
+				}
+			case stake.TxTypeSSRtx:
+				// All outputs for SSRtx tx are only spendable
+				// after coinbase maturity many blocks.
+				if !coinbaseMatured(w.chainParams, details.Height(), tipHeight) {
+					continue
+				}
+			}
+
+			// Exclude locked outputs from the result set.
+			if w.LockedOutpoint(&output.OutPoint.Hash, output.OutPoint.Index) {
+				continue
+			}
+
+			// Lookup the associated account for the output.  Use the
+			// default account name in case there is no associated account
+			// for some reason, although this should never happen.
+			//
+			// This will be unnecessary once transactions and outputs are
+			// grouped under the associated account in the db.
+			acctName := defaultAccountName
+			sc, addrs, _, err := txscript.ExtractPkScriptAddrs(
+				0, output.PkScript, w.chainParams, true) // Yes treasury
+			if err != nil {
+				continue
+			}
+			if len(addrs) > 0 {
+				acct, err := w.manager.AddrAccount(
+					addrmgrNs, addrs[0])
+				if err == nil {
+					s, err := w.manager.AccountName(
+						addrmgrNs, acct)
+					if err == nil {
+						acctName = s
+					}
+				}
```

Comment:
Silencing errors in general is not a great idea. Seems you want to specifically handle the case of the address not existing (which as you noted in the comment earlier, shouldn't happen), so you could check for `errors.NotExist` directly:

```go
acct, err := w.manager.AddrAccount(addrmgrNs, addrs[0])
if err == nil {
	var s string
	s, err = w.manager.AccountName(addrmgrNs, acct)
	if err == nil {
		acctName = s
	}
}
if err != nil && !errors.Is(err, errors.NotExist) {
	return err
}
```

Reply:
Should I also make this change here https://github.com/decred/dcrwallet/blob/master/wallet/wallet.go#L3633-L3641

Reply:
I'd do that as a separate PR.